packages/tpetra/core/src/Tpetra_Details_FixedHashTable_decl.hpp

```diff
@@ -127,7 +128,25 @@ class FixedHashTable {
   typedef Kokkos::View<const KeyType*, Kokkos::LayoutLeft, device_type> keys_type;
 
   //! Default constructor; makes an empty table.
-  KOKKOS_FUNCTION FixedHashTable ();
+  KOKKOS_FUNCTION FixedHashTable () :
+    minKey_ (::Kokkos::Details::ArithTraits<KeyType>::max ()),
+    maxKey_ (::Kokkos::Details::ArithTraits<KeyType>::is_integer ?
+             ::Kokkos::Details::ArithTraits<KeyType>::min () :
+             -::Kokkos::Details::ArithTraits<KeyType>::max ()),
+    minVal_ (::Kokkos::Details::ArithTraits<ValueType>::max ()),
+    maxVal_ (::Kokkos::Details::ArithTraits<ValueType>::is_integer ?
+             ::Kokkos::Details::ArithTraits<ValueType>::min () :
+             -::Kokkos::Details::ArithTraits<ValueType>::max ()),
+    firstContigKey_ (::Kokkos::Details::ArithTraits<KeyType>::max ()),
+    lastContigKey_ (::Kokkos::Details::ArithTraits<KeyType>::is_integer ?
+                    ::Kokkos::Details::ArithTraits<KeyType>::min () :
+                    -::Kokkos::Details::ArithTraits<KeyType>::max ()),
+    contiguousValues_ (true), // trivially
+    checkedForDuplicateKeys_ (true), // it's an empty table; no need to check
+    hasDuplicateKeys_ (false)
+  {
+  }
+
```

Comment:
@brian-kelley Thanks for the patch!

We are just wondering why the constructor of `FixedHashTable` needs to be inlined. On our side, it was working fine before (both Tpetra tests and our own tests passed).

Reply:
@romintomasetti Maybe you were building with relocatable device code, ``-DKokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE=ON``? I believe that lets you link to non-inline device code functions.

Reply:
@brian-kelley We are not using relocatable device code either... I don't know why I proposed changes that do not compile on our side either :/

I have update https://github.com/trilinos/Trilinos/pull/11218 a bit, taking into account your suggestions here, and applying a similar method to `FixedHashTable` as we did for `LocalMap` (i.e. define default value of each member in the declaration file).